### PR TITLE
fix: resize partition

### PR DESCRIPTION
**Pull request by sergelogvinov**
Resize function checks free space and grow it to the next partition.

need for [#4228](https://github.com/talos-systems/talos/issues/4228)

### CODE REVIEW

File: blockdevice/partition/gpt/gpt.go [hunk 1]
```diff
@@ -295,6 +295,78 @@ func (g *GPT) InsertAt(idx int, size uint64, setters ...PartitionOption) (*Parti
 	return partition, nil
 }
 
+// InsertAtOffset inserts partition starts from offset bytes.
+//
+// If idx == 0, it inserts new partition as the first partition, etc., idx == 1 as the second, etc.
+func (g *GPT) InsertAtOffset(idx int, offset, size uint64, setters ...PartitionOption) (*Partition, error) {
```
Comment (smira) [inline blockdevice/partition/gpt/gpt.go:301]: this seems to be really close to the `InsertAt` above ? should they be merged? do we need both? what is the difference?

Comment (sergelogvinov) [reply blockdevice/partition/gpt/gpt.go:301]: Yes, it can be merge, thank you. 
I'll try it.

Comment (smira) [reply blockdevice/partition/gpt/gpt.go:301]: I don't quite understand what it does though? does it enforce partition first LBA?

Comment (sergelogvinov) [reply blockdevice/partition/gpt/gpt.go]: Yep, I've added WithOffset options, which allows you to create partition starts from custom LBA offset (bytes offset -> LBA auto convert).

It helps to make resize unit-test, and for this https://github.com/talos-systems/talos/issues/4041

**Comment by Unix4ever** [comment]
@andrewrynhard and @smira. Are we ok that Talos is not going to occupy the whole disk? I'm not sure where ended up.

**Review by smira — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**